ci/infra/aws/load-balancer.tf

```diff
@@ -1,19 +1,55 @@
-resource "aws_elb" "kube_api" {
-  connection_draining       = false
-  cross_zone_load_balancing = true
-  idle_timeout              = 400
-  instances                 = aws_instance.control_plane.*.id
-  name                      = "${var.stack_name}-elb"
-  subnets                   = [aws_subnet.public.id]
+# A security group for the ELB so it is accessible via the web
+resource "aws_security_group" "elb" {
+  name        = "${var.stack_name}-elb"
+  description = "give access to kube api server"
+  vpc_id      = aws_vpc.platform.id
 
   tags = merge(
-    local.basic_tags,
+    local.tags,
     {
       "Name"  = "${var.stack_name}-elb"
-      "Class" = "ElasticLoadBalancer"
+      "Class" = "SecurityGroup"
     },
   )
 
+  # HTTP access from anywhere
+  ingress {
+    from_port   = 80
+    to_port     = 80
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  # HTTPS access from anywhere
+  ingress {
+    from_port   = 443
+    to_port     = 443
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  ingress {
```

Comment:
@dannysauer Correct me if i'm wrong but I don't think we should expose the API server too the web. this is an easy first step to securing them, making it so that only inside the cluster can directly hit the api server. I would be for removing this block.

Reply:
I'm not completely clear why we open 80/443 to begin with; nothing listens on those ports by default, right? Was that just a copy-paste from an example that was left behind all along?

Reply:
Also, yes, exposing the api server seems undesirable. I guess it's possible to control access to the whole vpc and that might be an acceptable control, but I honestly haven't read the entire config to see if that's being done. It still seems like it'd be preferable to elect a source IP as "management node" or identify a management network.

Reply:
Also, yes, exposing the api server seems undesirable. I guess it's possible to control access to the whole vpc and that might be an acceptable control, but I honestly haven't read the entire config to see if that's being done. It still seems like it'd be preferable to elect a source IP as "management node" or identify a management network.

Reply:
This is from original code. I just try to rearrange code by function. I will test to remove 80/443 to see if any side effect